src/routes/blog/ide-ux-bof-scipy-2024/+page.md

```diff
@@ -0,0 +1,141 @@
+In response, other attendees mentioned that the main challenge here was creating a way to indicate to consumers that data is expensive to generate a `repr`, which is where the costs really come in.
+
+Finally, a PyCharm user shared how Xarray/pandas, their extension system and data accessors were all opaque to its autocomplete, at least without further tagging or augmentation to the IDE / code.
+They expressed that to address this, IDEs should be better aware of extensions such as these.
+Others, including ourselves, mentioned in response that using customizable LSP servers can help make editors better aware of extensible syntaxes like this, which is indeed what Spyder and other IDEs do (and the Spyder team maintains the [de-facto standard pluggable Python implementation of](https://github.com/python-lsp/python-lsp-server)).
+
+While user needs here varied, there was a clear theme that users greatly value being able to interactively view, edit and analyze a diverse array of data types right within their IDE, as Spyder continues to particularly focus on.
+
+
+## Desktop vs web-based IDEs
+
+A popular point of discussion was the merits of desktop versus web-based interfaces and IDEs for science research, which saw some mixed opinions on the topic.
+
+Posing the question directly, we asked people to share their on a web-based application vs. a desktop application, and on a "desktop" app using a webview as its UI.
+One participant stated that a web browser can help them be effective for collaboration, but may not be as useful for single-user situations as a native application would be.
+Another had concerns with the usability of Zoom screen sharing, as sharing a desktop application just shared that application, while sharing a web-based IDE shared the user's entire web browser.
+A third suggested [vscode.dev](https://vscode.dev/), which runs in a browser and which can be self-hosted, but stated that Python support is not "fully on par" with the desktop version.
+A final point discussed was the possibility of running desktop IDEs inside a web-based container to improve security and ease of deployment, particularly using Python's new WASM support.
+
+On a related note, an attendee shared that they support a small statics group, and their biggest challenge is helping students learn scientific computing.
+They have a rather "rugged individualist culture", and feel they need to learn how to share knowledge with others, asking the audience how they could accomplish that with IDEs.
+This prompted others to related how that where they work, platforms for accessing data require a lot of authentication, and entail using JupyterHub as their default IDE/platform, with very little flexibility in the matter.
+They would like a platform for other IDEs to be able to connect to a shared JupyterHub environment, which confidential is exactly what Spyder is working on for Spyder 6 and beyond.
```

Comment:
```suggestion
They would like a platform for other IDEs to be able to connect to a shared JupyterHub environment, which is exactly what we are working on for Spyder 6.1 and beyond.
```

- I don't think it's necessary to say that this confidential when we're making it public.
- Avoid repeating Spyder.